src/page-templates/content-type/utils.ts

```diff
@@ -2,7 +2,120 @@ import { PillCategory } from '../../types/pill-category';
 import { ContentItem } from '../../interfaces/content-item';
 import { getAllContentItems } from '../../service/get-all-content';
 import { FilterItem } from '../../components/search-filters';
-import { itemsStyles } from '../../components/search-filters/styles';
+import { Tag } from '../../interfaces/tag';
+import { ShowcaseCardItem } from '../../components/showcase-card/types';
+import { languageToLogo } from '../../utils/language-to-logo';
+import { technologyToLogo } from '../../utils/technology-to-logo';
+import { productToLogo } from '../../utils/product-to-logo';
+import { ITopicCard } from '../../components/topic-card/types';
+
+interface TagWithCount extends Tag {
+    count: number;
+}
+
+interface L1TagWithCount extends TagWithCount {
+    l2s: Tag[];
+}
+
+export const getFeaturedLangProdTech = (
+    contentType: 'Tutorial' | 'Code Example',
+    content: ContentItem[]
+) => {
+    const aggregateSlug =
+        contentType === 'Tutorial' ? '/tutorials' : 'code-examples';
+    let languages: TagWithCount[] = [];
+    let technologies: TagWithCount[] = [];
+    let products: L1TagWithCount[] = [];
+    content.forEach(item => {
+        const languageTags = item.tags.filter(
+            tag => tag.type === 'ProgrammingLanguage'
+        );
+        languageTags.forEach(langTag => {
+            const existingTag = languages.find(
+                tag => tag.name === langTag.name
+            );
+            if (existingTag) {
+                existingTag.count += 1;
+            } else {
+                languages.push({ ...langTag, count: 1 });
+            }
+        });
```

Comment:
This code is repeated for `techTags` and `productTags` in addition to this. It can be moved out into a utility method (including the sorting part below) for ease of use and to reduce the repetition here.